src/tests/test_update.py

```diff
@@ -102,4 +105,71 @@ def test_statistics(self, mock):
         self.assertEqual(results['num_recompares'], 1)
         self.assertEqual(results['num_stored_events'], 1)
         self.assertEqual(results['num_skipped_events_skipped'], 1)
-        self.assertEqual(results['num_applied_applied'], 1)
\ No newline at end of file
+        self.assertEqual(results['num_applied_applied'], 1)
+
+    def test_gob_event_action(self, mock_event):
+        # setup event state
+        event = {
+            'sa_instance_state': '',
+            'version': '0.1',
```

Comment:
sa_instance, eventid, source_id, action and contents are not required in the event

Reply:
# setup event state
        event = {
            'version': '0.1',
            'catalogue': 'test_catalogue',
            'application': 'TEST',
            'entity': 'test_entity',
            'timestamp': datetime.datetime(2019, 1, 30, 18, 7, 7),
            'source': 'test'
        }

        # setup data
        data = {}

Also works and is a little bit cleaner